src/status_im/ui/screens/browser/events.cljs

```diff
@@ -58,6 +59,15 @@
      (-> (add-browser-fx cofx new-browser)
          (update-in [:db :browser/options] #(assoc % :browser-id (:browser-id new-browser)))))))
 
+(handlers/register-handler-fx
+ :update-browser-on-nav-change
+ [re-frame/trim-v]
+ (fn [{:keys [db now] :as cofx} [browser url loading]]
+   (let [new-browser (get-new-browser browser now)
```

Comment:
maybe we could move `get-new-browser` into model as well

Reply:
@flexsurfer Is it ok if we leave this one tabled? It could have been mentioned in previous code reviews and wasn't, and would also require renaming the model (and probably trigger another round of review). :)